taichi/runtime/gfx/runtime.cpp

```diff
@@ -470,6 +469,15 @@ void GfxRuntime::launch_kernel(KernelHandle handle, RuntimeContext *host_ctx) {
   // Record commands
   const auto &task_attribs = ti_kernel->ti_kernel_attribs().tasks_attribs;
 
+  if (args_buffer) {
```

Comment:
Do I understand correctly that: if `args_buffer` is allocated on the device, use that; otherwise assume this is using push constants?

Reply:
Yes. But actually the other way around, args_buffer is allocated iff we are not using push constants (i.e. when the device supported maximum push constants size is smaller than the args)